source/_docs/scripts.markdown

```diff
@@ -598,6 +598,22 @@ script:
           message: "Skipped cleaning, someone is home!"
 ```
 
+<div class='note'>
+
+When testing for a state, you cannot use a template for the `entity_id` as it has to be a constant.
```

Comment:
I'm not following, the forum post is about a condition, not if/else. 

Rhe condition is documented else where? What has this to do with the if/else statement?

Reply:
The title of the post is "Automation with for_each and if" so it is clear to me the post is about using if in a specific case. The relevance becomes especially apparent when reading the last two responses to that post. I had a similar situation where I had an if within a for and wanted to check the state of the item in question. However as the last answer makes clear: it is not possible to use a jinja template for the entity_id in an if entirely.

Though this applies generally to conditions and triggers which is not explicitly marked in the documentation for if. It would appear to me that this is, what is referred to [limited templates](https://www.home-assistant.io/docs/configuration/templating/#limited-templates) elsewhere. However there are multiple issues with the documentation of that specific behaviour.

Reply:
> The title of the post is "Automation with for_each and if"

It is not? This is a separate/new chapter.

Reply:
![screenshot of the forum post](https://user-images.githubusercontent.com/9216782/185801543-ca26de69-5f3e-458c-8b9a-b198ed8c9e3d.png)

Correct me if I am wrong, but isn't that the title of the post?

Reply:
Sure, but I am still not sure what this has to do with this chapter of the documentation in any way. I've read the forum post, which is about using a template for the `entity_id` in a state condition. That has nothing to do with if-actions. Sure, the if action accepts conditions, but your comment and question are specifically about the state condition, not the if action.

Reply:
So the condition of an action and of an if are exactly the same? Sorry, if this got a little heated but spending days on fixing such basic issues in my config has been really draining.

What follows is a fairly long insight into my way of thinking and interacting with Home Assistant. It is not meant as a criticism. Feel free to skip to the Tl;Dr.

For me the documentation in general is very confusing and I think this issue is just a manifestation of that. As a (admittedly junior) developer I interact often with documentation. Even though Home Assistant is somewhat larger compared to the biggest  frameworks I worked with (like Angular or Laravel) I struggle to get even basic stuff done. My guess is that this is due to the structure of the documentation. It is fairly fragmented, doesn't have a navigable over view and is not very searchable.
For example my case: I want my motion sensors to control some rooms and adjust the brightness the lights are turned on with depending of time of day.
Because I want something to be triggered automatically I need an Automation. Though because I want this for multiple rooms I and I don't want to repeat code I need a blueprint. Automations are written in YAML with Jinja2 templating and the schema for actions are called Scripting. 
While I have some experience with YAML and Jinja is not very hard to understand the basics, thought for some details I have the documentation open. Though as Home Assistant supplements the Jinja in-built functions I also need the Home Assistant Template documentation open. As I am writing a blueprint for an automation I also need the Automation and Blueprint documentation open. Then for the actions I need the scripting documentation. And then some documentation for auxiliary things like the Selectors in Blueprints.

With now ~7 tabs open and constantly switching between them it is often unclear what aspect belongs to what documentation, especially as they all interweave. To come back to my original problem: it came to me as a total surprise that the `entity_id` of an `if` isn't part of the `if`. It isn't even part of scripting but instead belongs to the Template documentation.

Now with all that said: this isn't supposed to be a rant or a critique of the countless hours of work you and other volunteers have spent not only making but also documenting and supporting the brilliant ecosystem that is Home Assistant. Though with the almost weeks as a somewhat experienced developer writing what is supposed to be a simple automation I feel like I am doing something wrong or that my use case is plainly too niche. I would be extremely thankful if you have the time and motivation to point out issues with my approach though I can very much understand if you don't have the time or plainly the motivation to deal with my personal problems.

Tl;Dr: I have trouble with the documentation and this specific behaviour has stumped me and others. While I think I understand what you mean and in that regard agree with you, I would still recommend _some_ change to the documentation.

As for the PR: I would still favour at least expanding the documentation in Templating to more explicitly document where and where not (most) templating can be used. I.e. not in conditions and triggers. Or even more explicitly that they can not be used as `entity_id` in if-actions, conditions for actions etc. Also a notice in the "Important Template Rules" warning, that not all of templating can be used everywhere would be beneficial as I regard it as a fairly heavy limitation of templating.

Feel free to resolve the conversation and/or close the PR. I don't think I would do a PR for the proposed changes unless there is prior agreement that the change is warranted.